src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/commands/mobile/set_interior_vehicle_data_request.cc

```diff
@@ -265,13 +421,36 @@ void SetInteriorVehicleDataRequest::on_event(
 
 const smart_objects::SmartObject& SetInteriorVehicleDataRequest::ControlData(
     const smart_objects::SmartObject& module_data) {
-  const std::string module = ModuleType();
+  const std::string module_type = ModuleType();
+  std::map<std::string, std::string> data_mapping =
+      GetModuleTypeToDataMapping();
+  for (const auto& data : data_mapping) {
+    if (data.first == module_type) {
+      return module_data[data.second];
+    }
+  }
+  NOTREACHED();
+  return module_data[0];
+}
 
-  if (enums_value::kRadio == module) {
-    return module_data[message_params::kRadioControlData];
-  } else {
-    return module_data[message_params::kClimateControlData];
+void SetInteriorVehicleDataRequest::CheckAudioSource(
+    const smart_objects::SmartObject& audio_data) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (mobile_apis::PrimaryAudioSource::MOBILE_APP !=
+      audio_data[message_params::kSource].asInt()) {
+    if (!audio_data.keyExists(message_params::kKeepContext) ||
+        !audio_data[message_params::kKeepContext].asBool()) {
+      if (mobile_apis::PrimaryAudioSource::MOBILE_APP ==
+          application_manager_.get_current_audio_source()) {
+        app_mngr::ApplicationSharedPtr app =
+            application_manager_.application(connection_key());
+        application_manager_.ChangeAppsHMILevel(
+            app->app_id(), mobile_apis::HMILevel::eType::HMI_BACKGROUND);
+      }
+    }
   }
+  application_manager_.set_current_audio_source(
```

Comment:
shall this be set only after sdl forward the request to HMI and get a successful response?

Reply:
@yang1070 Fixed 7211a368690865058ace2eff1811baa788faf801